.github/workflows/terraform-cicd.yaml

```diff
@@ -0,0 +1,85 @@
+name: 'Terraform-Workflow'
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    types: [opened, edited, reopened, ready_for_review, review_requested]
```

Comment:
@sjoshi-jpl similar question to other TF PRs, do we really want this on branches/PRs or do we just want this on merges to main?

Reply:
@jordanpadams yes, its generally a good practice to run at least `terraform plan` before merging to main because if there are any issues within the terraform code itself, it won't be caught until the code is merged to main.

Additionally, when a developer is testing code changes (for their specific branch), its helpful to be able to view the output of the run within GitHub in order to troubleshoot any issues. So they don't create a PR before those errors are resolved.

If you'd still like to limit the number of times tf plan runs, I can change it to only execute during a push to main and when a PR for main is created.